test/runtests.jl

```diff
@@ -8,1930 +8,1706 @@ n = 35
+            @test host_dot ≈ cuda_dot1
+            @test host_dot ≈ cuda_dot2
+            @test host_dot ≈ (d_A.'*d_B)[1]
+        end
+    end
+end
+
+@testset "dotc" begin
+    @testset for elty in [Complex64, Complex128]
+        A = rand(elty, m)
+        B = rand(elty, m)
+        @test ndims(A) == 1
+        @test ndims(B) == 1
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B = CudaArray(B)
+        cuda_dot1 = CUBLAS.dotc(n1,d_A,1,d_B,1)
+        cuda_dot2 = CUBLAS.dotc(d_A,d_B)
+        host_dot = A'*B
+        if VERSION < v"0.6.0-pre"
+            @test host_dot[1] ≈ cuda_dot1
+            @test host_dot[1] ≈ cuda_dot2
+            @test host_dot ≈ (d_A'*d_B)
+        else
+            @test host_dot ≈ cuda_dot1
+            @test host_dot ≈ cuda_dot2
+            @test host_dot ≈ (d_A'*d_B)[1]
+        end
+    end
+end
+
+@testset "nrm2" begin
+    @testset for elty in [Float32, Float64, Complex64, Complex128]
+        A = rand(elty, m)
+        @test ndims(A) == 1
+        n1 = length(A)
+        d_A = CudaArray(A)
+        cuda_nrm2_1 = CUBLAS.nrm2(n1,d_A,1)
+        cuda_nrm2_2 = CUBLAS.nrm2(d_A)
+        cuda_nrm2_3 = norm(d_A)
+        host_nrm2 = norm(A)
+        @test host_nrm2 ≈ cuda_nrm2_1
+        @test host_nrm2 ≈ cuda_nrm2_2
+        @test host_nrm2 ≈ cuda_nrm2_3
+    end
+end
+
+@testset "asum" begin
+    @testset for elty in [Float32, Float64, Complex64, Complex128]
+        A = rand(elty, m)
+        @test ndims(A) == 1
+        n1 = length(A)
+        d_A = CudaArray(A)
+        cuda_asum1 = CUBLAS.asum(n1,d_A,1)
+        cuda_asum2 = CUBLAS.asum(d_A)
+        host_asum = sum(abs.(real(A)) + abs.(imag(A)))
+        @test host_asum ≈ cuda_asum1
+        @test host_asum ≈ cuda_asum2
+    end
+end
+
+@testset "axpy!" begin
+    # test axpy!
+    function test_axpy!_1(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        CUBLAS.axpy!(n1,alpha,d_A,1,d_B1,1)
+        B1 = to_host(d_B1)
+        host_axpy = alpha*A + B
+        @test host_axpy ≈ B1
+    end
+    test_axpy!_1(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_1(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_1(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_1(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))
+
+    function test_axpy!_2(alpha,A,B)
+        @test length(A) == length(B)
+        n1 = length(A)
+        d_A = CudaArray(A)
+        d_B1 = CudaArray(B)
+        CUBLAS.axpy!(alpha,d_A,d_B1)
+        B1 = to_host(d_B1)
+        host_axpy = alpha*A + B
+        @test host_axpy ≈ B1
+    end
+    test_axpy!_2(2.0f0,rand(Float32,m),rand(Float32,m))
+    test_axpy!_2(2.0,rand(Float64,m),rand(Float64,m))
+    test_axpy!_2(2.0f0+im*2.0f0,rand(Complex64,m),rand(Complex64,m))
+    test_axpy!_2(2.0+im*2.0,rand(Complex128,m),rand(Complex128,m))
+
+    #=function test_axpy!_3(alpha,A,B)
```

Comment:
Same here. Would be nice to have a comment explaining why they need to be commented out